### PR TITLE
Allow to pass message as concatenated strings

### PR DESCRIPTION
**Pull request by victor-homyakov**

### CODE REVIEW

**Comment by Mavrin** [comment]
Lgtm. But would be nice if you add something to readme file, especially corner case `intl.formatMessage('crack' + " me')");`

**Comment by Mavrin** [comment]
lgtm

**Comment by Metallium** [comment]
lgtm

File: lib/lex.js [hunk 1]
```diff
@@ -1,60 +1,91 @@
...
-        if (c === quoteChar && lexeme.charAt(i - 1) !== '\\') break;
-        if (quoteChar) str += c;
+    if (str.length === len) {
+        return str;
     }
 
-    return str;
+    var str2 = extractFromExpression(quoteChar + message + quoteChar);
+    if (str2) {
+        /*console.warn('Message is extracted from AST:',
```
Comment (victor-homyakov) [inline lib/lex.js]: Done

File: lib/lex.js [hunk 2]
```diff
@@ -1,60 +1,91 @@
...
-                lexer = new Lexer(function () {
+                lexer = new Lexer(function() {
                 });
-            config.scopeRules.forEach(function (rule) {
-                lexer.addRule(rule, function (lexeme, value) {
+            config.scopeRules.forEach(function(rule) {
+                lexer.addRule(rule, function(lexeme, value) {
                     if (value) {
+                        if (scope) {
+                            console.error('Scope is already defined as "' + scope +
```
Comment (victor-homyakov) [inline lib/lex.js]: Removed console output in this PR

**Pull request merged**